src/peft/tuners/lora.py

```diff
@@ -301,16 +277,47 @@ def _create_new_module(self, lora_config, adapter_name, target):
 
         return new_module
 
+    def _yield_matching_target_keys(self, model, lora_config: LoraConfig):
+        """For the given model and config, yield all module names that match."""
+
+        # normalize the parameters
+        target_modules = lora_config.target_modules
+        if isinstance(target_modules, str):
+            target_modules = [target_modules]
```

Comment:
Hello, this is incorrect. Here, if `target_modules` is str, it needs to be treated as a regex as mentioned in the LoraConfig param help metadata, e.g., `'.*decoder.*(SelfAttention|EncDecAttention).*(q|v)$'`